src/event-store/services/event-store.service.ts

```diff
@@ -0,0 +1,490 @@
+  AppendToStreamOptions,
+  GetStreamMetadataResult,
+  ReadStreamOptions,
+  SetStreamMetadataOptions,
+} from '@eventstore/db-client/dist/streams';
+import { ReadableOptions } from 'stream';
+import {
+  PersistentSubscription,
+  persistentSubscriptionSettingsFromDefaults,
+  ResolvedEvent,
+} from '@eventstore/db-client';
+import { StreamMetadata } from '@eventstore/db-client/dist/utils/streamMetadata';
+import { PersistentSubscriptionSettings } from '@eventstore/db-client/dist/utils';
+import { isNil } from '@nestjs/common/utils/shared.utils';
+import { Client } from '@eventstore/db-client/dist/Client';
+import { EVENT_STORE_CONNECTOR } from './event-store.constants';
+import {
+  CreateContinuousProjectionOptions,
+  CreateOneTimeProjectionOptions,
+  CreateTransientProjectionOptions,
+  GetProjectionStateOptions,
+} from '@eventstore/db-client/dist/projections';
+import { DeletePersistentSubscriptionOptions } from '@eventstore/db-client/dist/persistentSubscription';
+import {
+  PERSISTENT_SUBSCRIPTION_ALREADY_EXIST_ERROR_CODE,
+  PROJECTION_ALREADY_EXIST_ERROR_CODE,
+} from './errors.constant';
+import EventHandlerHelper from './event.handler.helper';
+import IEventsAndMetadatasStacker, {
+  EVENTS_AND_METADATAS_STACKER,
+} from '../reliability/interface/events-and-metadatas-stacker';
+import EventBatch from '../reliability/interface/event-batch';
+
+@Injectable()
+export class EventStoreService implements OnModuleInit, IEventStoreService {
+  private logger: Logger = new Logger(this.constructor.name);
+  private persistentSubscriptions: PersistentSubscription[];
+
+  private isOnError = false;
+  private isTryingToConnect = true;
+  private isTryingToWriteEvents = false;
+  private isTryingToWriteMetadatas = false;
+
+  constructor(
+    @Inject(EVENT_STORE_CONNECTOR)
+    private readonly eventStore: Client,
+    @Inject(EVENT_STORE_SUBSYSTEMS)
+    private readonly subsystems: IEventStoreSubsystems,
+    @Inject(EVENTS_AND_METADATAS_STACKER)
+    private readonly eventsStacker: IEventsAndMetadatasStacker,
+    @Optional() private readonly eventBus?: ReadEventBus,
+  ) {}
+
+  public async onModuleInit(): Promise<void> {
+    return await this.connect();
+  }
+
+  private async connect(): Promise<void> {
+    try {
+      if (this.subsystems.projections)
+        await this.assertProjections(this.subsystems.projections);
+      if (this.subsystems.subscriptions)
+        this.persistentSubscriptions =
+          await this.subscribeToPersistentSubscriptions(
+            this.subsystems.subscriptions.persistent,
+          );
+
+      this.isOnError = false;
+      this.isTryingToConnect = false;
+      this.logger.log(`EventStore connected`);
+    } catch (e) {
+      this.isTryingToConnect = true;
+      await this.retryToConnect();
+    }
+  }
+
+  private async retryToConnect(): Promise<void> {
+    this.logger.log(`EventStore connection failed : trying to reconnect`);
+    setTimeout(async () => await this.connect(), 1000);
```

Comment:
Maybe adding an exponential backoff strategy something like 'wait for (2^retries * 100) milliseconds' and allowing a maximum of reconnection retries will be safer, what do you think about this?

Reply:
As discussed with the @Vinceveve, kubernetes should restart the eventstore in about 10 min, but we cannot be sure of the reason of why there is no more connection to es : internet connection issue, system issue or eventstore issue. And in each of these cases, the waiting time could be different. It seemed to me that 1s was the right "middle way". 
Furthermore, if I stop trying to reconnect after n attempts, I might never reconnect at all.

Reply:
exponential backoff strategy, would be a good idea, if using rxjs, you can use `retryBackoff` operator